src/python/pants/engine/fs.py

```diff
@@ -188,6 +188,13 @@ def is_empty(self):
     return self == EMPTY_SNAPSHOT
 
 
+@dataclass(frozen=True)
+class SnapshotSubset:
+  """A request to create a subset of a snapshot."""
+  directory_digest: Digest
+  includes: PathGlobs
```

Comment:
`s/includes/globs` or `path_globs`. `PathGlobs` support excludes and this will get even easier to express with https://github.com/pantsbuild/pants/pull/9025.

It's also very useful to a rule author to say `get me this whole dir, but not this one file 'foo'`.